### DISCUSSION TITLE
GitHub claims my password is used on other sites...

### DISCUSSION THREAD

**Discussion by galmok** [community/community#33353] [Other Feature Feedback, Questions, & Ideas]
GitHub claims my password is used on other sites, but how would they know unless they are able to compare my password or a hash of it with other sites? Storing passwords using a simple hash is not really secure as it needs a salt and even better, using PBKDF2. This makes comparing passwords impossible as the same password stored multiple times wont be similar in any way.

So, from that warning I can only read that GitHub and the other unmentioned sites use an inferior password storing technique. And they complain about my password?...

**Comment by airtower-luna** [comment]
Your password must be sent unhashed when you change it (or log in), so it can be hashed and salted and then stored in the database (or compared to what's in the database). So any service where you log in with a password has your unhashed password *temporarily*, any service with decent security just won't store it that way.

At that point it's easy (and common practice) to compare it against passwords from one of the many leaks that became public, or compilations of those. :wink:

**Reply by galmok** [reply]
But my password isn't leaked. I have had a leaked password and the message reflected that. My current password just is the same as on another platform. I don't know how they know that unless they password storing is flawed and allows comparison of passwords, something PBKDF2 explicitly prevents (unless the salt is identical on both platforms, but it should be random...).

**Reply by airtower-luna** [reply]
When you log in or change the password your browser sends the plain password (over an encrypted connection that ends at a GitHub server), the hashing happens on the server. So they can easily compare it to whatever lists they use then.

I don't know *what* they compare with.
